oddt/docking/AutodockVina.py

```diff
@@ -324,6 +338,51 @@ def predict_ligands(self, ligands):
         return self.score(ligands)
 
 
+def write_vina_pdbqt(mol, directory, flexible=True, name_id=None):
+    """Write single PDBQT molecule to a given directory. For proteins use
+    `flexible=False` to avoid encoding torsions. Additionally an name ID can
+    be appended to a name to avoid conflicts.
+    """
+    if name_id is None:
+        name_id = ''
+
+    # We expect name such as 0_ZINC123456.pdbqt or simply ZINC123456.pdbqt if no
+    # name_id is specified. All non alpha-numeric signs are replaced with underscore.
+    mol_file = ('_'.join(filter(None, [str(name_id),
+                                       re.sub('[^A-Za-z0-9]+', '_', mol.title)]
+                                )) + '.pdbqt')
+    # prepend path to filename
+    mol_file = os.path.join(directory, mol_file)
+
+    if (hasattr(oddt.toolkits, 'ob') and
+            isinstance(mol, oddt.toolkits.ob.Molecule)):
+        if flexible:
+            # auto bonding (b), perserve atom names (n) indices (p) and Hs (h)
+            kwargs = {'opt': {'b': None, 'p': None, 'h': None, 'n': None}}
+        else:
+            # for proteins write rigid mol (r) and combine all frags in one (c)
+            kwargs = {'opt': {'r': None, 'c': None, 'h': None}}
+
+    else:
+        kwargs = {'flexible': flexible}
+
+    # HACK: fix OB 2.3.2 PDBQT bugs
+    if (hasattr(oddt.toolkits, 'ob') and
```

Comment:
I can add it, but this does not harm OB 2.4+, since these tags do not occur for rigid molecule there. Although performance wise it makes sense to skip it.